mod.ts

```diff
@@ -134,6 +134,11 @@ export class Udd {
       newVersion = compatible[0];
     }
 
+    // Put the version token back where it was at the prefix since we moved it to fragment 
+    if (newFragmentToken != undefined) {
+      newVersion = `${newFragmentToken}${newVersion}`
```

Comment:
what do these lines do?

Reply:
Take for example this dependency: `npm:foo@~3.0.0`. 

[This line of code](https://github.com/levibostian/deno-udd/blob/694beac43b4f151f45fcb928b0c4dfcfca0a0587/mod.ts#L91-L94) transforms that into `npm:foo@3.0.0#~`. Then udd will perform the logic to find the version to upgrade to. Let's say udd says that 4.1.0 is the newest version. 

After udd has found the version to upgrade to, we need to convert the syntax back to the `npm:` syntax originally used. Which is partially what this line of code does. This line of code will take `npm:foo@4.1.0#~` and turn it into `npm:foo@~4.1.0#~`. Then, when `url.at(...)` is called to write the new changes to the file, `npm:foo@~4.1.0` is returned from `url.at()` to bring back our original `npm` syntax.